src/state/reducer.ts

```diff
@@ -0,0 +1,36 @@
+import { GlobalState } from './state';
+import type { ArweaveTransactionId, ArNSContractState } from '../types';
+
+export type Action =
+  | { type: 'setWalletAddress'; payload: ArweaveTransactionId }
+  | { type: 'setGateway'; payload: string }
+  | { type: 'setArnsContractState'; payload: ArNSContractState }
+  | { type: 'setErrors'; payload: Array<Error> };
+
+export const reducer = (state: GlobalState, action: Action): GlobalState => {
+  switch (action.type) {
+    case 'setWalletAddress':
+      return {
+        ...state,
+        connectedWallet: action.payload,
+      };
+    case 'setGateway':
+      return {
+        ...state,
+        gateway: action.payload,
+      };
+    case 'setArnsContractState':
+      return {
+        ...state,
+        arnsSourceContract: action.payload,
+      };
+    case 'setErrors':
+      return {
+        ...state,
+        errors: action.payload,
+      };
```

Comment:
we're going to want to be able to push errors and remove errors from the array, so for now, let's move this
```suggestion
```